Add java solution to clock tree

nathangong opened the pull request.

brad182 commented on solutions/usaco-1016.mdx:
```diff
@@ -0,0 +1,180 @@
+
+		// same mod means you can start anywhere
+		if ((sum0 % 12) == (sum1 % 12)) {
+			io.println(n);;
```
two semicolons
```suggestion
			io.println(n);
```

brad182 commented on solutions/usaco-1016.mdx:
```diff
@@ -0,0 +1,180 @@
+		io.close();
+	}
+
+	static void dfs(int i, int color, int par) {
```
```suggestion
	static void dfs (int i, int color, int par) {
```

nathangong replied:
I think it's convention to not leave a space in between the method name and parameters?
![image](https://user-images.githubusercontent.com/8647920/125383473-4fc0ff00-e34c-11eb-8e3f-4e98d108fe04.png)

brad182 replied:
oh

brad182 commented on solutions/usaco-1016.mdx:
````diff
@@ -0,0 +1,180 @@
+```java
+public class ClockTree {
+	static int[] clocks;
+	static List<ArrayList<Integer>> adj;
````
probably could be an array of arraylists, but doesn't matter that much

nathangong replied:
Yeah I did this cause we're suposed to do polymorphism, but it also seems a bit weird so idk

brad182 commented on solutions/usaco-1016.mdx:
```diff
@@ -0,0 +1,180 @@
+	static List<ArrayList<Integer>> adj;
+	static int sum0 = 0, sum1 = 0, nodes0 = 0, nodes1 = 0;
+
+	public static void main(String[] args) throws IOException {
```
```suggestion
	public static void main (String[] args) throws IOException {
```

brad182 reviewed (changes requested):
minor changes

brad182 commented:
wait you changed it to arraylists of arraylists i was saying it was fine, just commented that I usually do array of arraylists instead

nathangong commented:
Yeah I think ArrayList of ArrayLists makes more sense and is less confusing, and unfortunately Java doesn't allow making a List of Lists

HarryWangATX reviewed (changes requested):
time complexity

HarryWangATX commented:
> I tried to do List<List> but it didn't work, so the only other option would be List<ArrayList> but I think that may be a bit weird/confusing

wdym by doesn't work? that syntax is valid.

nathangong commented:
Java throws this error when I try to do
```java List<List<Integer>> adj = new ArrayList<ArrayList<Integer>>();```
![image](https://user-images.githubusercontent.com/8647920/125656804-02e54c94-5a4f-4885-abf2-272386e299f1.png)

nathangong commented:
Wait I think ```List<List<Integer>> = new ArrayList<>();``` might work

HarryWangATX reviewed (approved):
code looks good, havent looked at explanation yet, core needs to review it too.

nathangong commented:
The explanation was approved a while back, the reason why it looks like I added everything was because I changed the file name to match the new naming standards
